src/orion/algo/random.py

```diff
@@ -55,17 +55,15 @@ def suggest(self, num=1):
         .. note:: New parameters must be compliant with the problem's domain
            `orion.algo.space.Space`.
         """
+        if num is None:
+            num = 1
+
         points = []
-        point_ids = set(self._trials_info.keys())
-        i = 0
         while len(points) < num:
-            new_point = self.space.sample(
-                1, seed=tuple(self.rng.randint(0, 1000000, size=3))
-            )[0]
-            point_id = infer_trial_id(new_point)
-            if point_id not in point_ids:
-                point_ids.add(point_id)
+            seed = tuple(self.rng.randint(0, 1000000, size=3))
+            new_point = self.space.sample(1, seed=seed)[0]
+            if not self.has_suggested(new_point):
+                self._trials_info[self.get_id(new_point)] = (new_point, None)
```

Comment:
Seems all algorithms will need to append new suggestions into `_trials_info`, can we move it to the base algorithm `suggest` method?

Reply:
Yes we should turn in into a method instead of duplicating it everywhere. I will do this.